nectar/downloaders/threaded.py

```diff
@@ -273,119 +273,128 @@ def _fetch(self, request, session):
-            report.error_msg = e.args[2]
-            report.error_report['response_code'] = e.args[1]
-            report.error_report['response_msg'] = e.args[2]
-            report.download_failed()
-
-        except Exception as e:
-            _logger.exception(e)
-            report.error_msg = str(e)
-            report.download_failed()
+                if ignore_encoding or self.config.stream:
+                    chunks = self.chunk_generator(response.raw, self.buffer_size)
+                else:
+                    chunks = response.iter_content(self.buffer_size)
+
+                for chunk in chunks:
+                    if self.is_canceled or request.canceled:
+                        raise DownloadCancelled(request.url)
+
+                    file_handle.write(chunk)
+
+                    bytes_read = len(chunk)
+                    report.bytes_downloaded += bytes_read
+
+                    now = datetime.datetime.now()
+
+                    if now - last_update_time >= progress_interval:
+                        last_update_time = now
+                        self.fire_download_progress(report)
+
+                    with self._bytes_lock:
+                        if now - self._time_bytes_this_second_was_cleared >= ONE_SECOND:
+                            self._bytes_this_second = 0
+                            self._time_bytes_this_second_was_cleared = now
+                        self._bytes_this_second += bytes_read
+
+                    if max_speed is not None and self._bytes_this_second >= max_speed:
+                        # it's not worth doing fancier mathematics than this, very
+                        # fine-grained sleep times [1] are not honored by the system
+                        # [1] for example, sleeping the remaining fraction of time
+                        # before this second is up
+                        time.sleep(0.5)
+
+                # guarantee 1 report at the end
+                self.fire_download_progress(report)
+
+            except SkipLocation:
+                _logger.debug("Skipping {url} because {netloc} could not be reached.".format(
+                    url=request.url, netloc=netloc)
+                )
+                report.download_skipped()
+
+            except requests.ConnectionError as e:
+                if nretry < DEFAULT_TRIES - 1:
+                    continue
+                _logger.error(_('Skipping requests to {netloc} due to repeated connection'
+                                ' failures: {e}').format(netloc=netloc, e=str(e)))
+                self.failed_netlocs.add(netloc)
+                report.download_connection_error()
+
+            except requests.Timeout:
+                if nretry < DEFAULT_TRIES - 1:
+                    continue
+                """
+                Handle a timeout differently than a connection error. Do not add
+                to failed_netlocs so that a new connection can be attempted.
```

Comment:
based on the docs string a new connection should be established if hitting timeout error, is the re-try logic needed here as well?

Reply:
I'm not sure what error will be produced if server reset connection during stream so I assumed it couldn't hurt to retry here as well. I'm fine if removing the retry process if you're against this

Reply:
I think my reply above have answer this question.